perl.h

```diff
@@ -5427,6 +5427,25 @@ EXTCONST char *const PL_phase_names[] = {
     "END",
     "DESTRUCT"
 };
+
+/*
+=for apidoc_section $utility
+
+=for apidoc Amd|char *const|perl_phase_name|enum perl_phase
+
+Returns the given phase's name as a NUL-terminated string.
+
+For example, a warning that includes the current interpreter phase
+might look thus:
+
+    const char* phase_name = perl_phase_name(PL_phase);
+    warn("This is weird. (Perl phase: %s)", phase_name);
+
+=cut
+*/
+
+#define perl_phase_name(phase) (PL_phase_names[phase])
```

Comment:
It is very rare (I want to say "never the case", but maybe there are a few) that visible macros take the `perl_` prefix on them. I'd suggest this should be called simply `phase_name()`.

Reply:
@leonerd I was hoping to retain the “namespacing” that `PL_phase` has. `PL_` is a no-no, though.

Things like form() and mess() are hard to search for. Much easier if there’s some sort of prefix on there.

I’ll change if you still think it better, but given that there are interfaces concerning the interpreter that have the `perl_` prefix it seems a bit more helpful to me to leave that on.

Reply:
`perl_` has been used in the code in the past to signify "here's our unique twist of an external thing".  It does have meaning and using it here should be avoided.

Reply:
@smpeters `perl_` in the API (cf. `perldoc api`) seems to indicate something that pertains to an interpreter, e.g., `perl_clone`, `perl_alloc`, etc.

I’ve gone and removed the prefix. I still think it would be more sensible left in, but it’s not worth the back-and-forth.

Thanks!